File: tests/integration/non-null-debug-test.ts
// Simple debug test for non-null assertion
function testSimple(value: string | null): number {
  // Test without non-null assertion first
  const valueLength = value ? value.length : 0;
  return valueLength;
}

function testNonNull(value: string | null): number {
  // Test with non-null assertion
  return value!.length;
}

function main(): void {
  const _test1 = testSimple("hello");
  const _test2 = testNonNull("world");
}

main();

File: tests/integration/non-null-debug-test/non-null-debug-test.cpp
#include "non-null-debug-test.h"

using namespace js;

js::number testSimple(js::typed::Nullable<js::string> value) {
    const js::any valueLength = (js::null ? js::null : js::null);
    return valueLength;
}

js::number testNonNull(js::typed::Nullable<js::string> value) {
    return value.length();
}

void main() {
    const js::any _test1 = testSimple("hello"_S);
    const js::any _test2 = testNonNull("world"_S);
}
